docs/guides/upgrade-apps-without-downtime.md

```diff
@@ -0,0 +1,182 @@
+accordingly: it will regenerate NGINX configuration and then send a reload signal to the NGINX master process to apply
+the new configuration (this process is further explained in [NGINX documentation][reconfiguration]). We call such an
+operation a reload. Because client requests do not get dropped during a reload, it is considered graceful.
+
+[reconfiguration]:https://nginx.org/en/docs/control.html?#reconfiguration
+
+
+> See also the [Architecture doc](/docs/architecture.md) to learn more about NGINX Kubernetes Gateway architecture.
+
+A subset of all possible configuration changes is changes to Endpoints, which are the most frequent changes during
+an application upgrade. How NGINX handles them also prevents downtime.
+
+### Adding or Removing Endpoints
+
+During an upgrade of an application, Kubernetes starts the pods of the new version and brings down the old ones. It also
+deletes and creates the corresponding Endpoints. NGINX Kubernetes Gateway sees the changes to the Endpoints by watching
+for changes to the corresponding [EndpointSlices][endpoint-slices].
+
+[endpoint-slices]:https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/
+
+In NGINX configuration, a Service is represented as an [upstream][upstream], and an Endpoint as an
+[upstream server][upstream-server].
+
+[upstream]:https://nginx.org/en/docs/http/ngx_http_upstream_module.html#upstream
+
+[upstream-server]:https://nginx.org/en/docs/http/ngx_http_upstream_module.html#server
+
+Let's consider two cases:
+
+- If an Endpoint is added, NGINX Kubernetes Gateway will add an upstream server that corresponds to the Endpoint to
+- NGINX and reload NGINX. After that, NGINX will start proxying traffic to it.
+- If an Endpoint is removed, NGINX Kubernetes Gateway will remove the corresponding upstream server from NGINX. After
+  a reload, NGINX will stop proxying traffic to it. However, it will finish proxying any pending requests to that
+  server,
+  which prevents downtime.
+
+As a result, as long you have more than one ready Endpoint, the clients should not experience any downtime during
+an upgrade.
+
+> It is a good practice to configure a [Readiness probe][readiness-probe] in the Deployment so that a pod can advertise
+> when it is ready to receive traffic. Note that NGINX Kubernetes Gateway will not add any Endpoint to NGINX that is not
+> ready.
+
+[readiness-probe]:https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/
+
+The upgrade methods from the next sections will change upstream servers in NGINX.
+
+## Rolling Deployment Upgrade
+
+To start a [rolling Deployment upgrade][rolling-upgrade], you update the Deployment to use the new version tag of
+the application. As a result, Kubernetes will terminate the pods with the old
+version and create new ones, ensuring, by default, that some number of pods always stays available during this process.
```

Comment:
```suggestion
version and create new ones. This ensures that some number of Pods always stays available during this process.
```

Reply:
address this in https://github.com/nginxinc/nginx-kubernetes-gateway/pull/944/commits/3622223d78183c328c4424bac13443389efdc961 I retained by default, because this behavior could be changed if the default .spec.strategy.type is changed from to the Recreate